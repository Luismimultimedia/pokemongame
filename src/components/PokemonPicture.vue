<template>
    <div class="pokemon-container">
        <img
            v-if="!showPokemon"
            class="hidden-pokemon"
            :src="imgSrc"
            alt="pokemon"
        />
        <img v-else class="fade-in" :src="imgSrc" alt="pokemon" />
    </div>
</template>

<script>
export default {
    name: "PokemonPicture",
    props: {
        pokemonId: {
            default: 1,
            type: Number,
            required: true,
        },
        showPokemon: {
            default: false,
            type: Boolean,
            required: true,
        },
    },
    computed: {
        imgSrc() {
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
        },
    },
};
</script>

<style scoped>
img {
    height: 200px;
    position: absolute;
    left: 0;
    right: 0;
    margin-left: auto;
    margin-right: auto;
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}

.hidden-pokemon {
    filter: brightness(0);
}

.pokemon-container {
    height: 200px;
}
</style>
