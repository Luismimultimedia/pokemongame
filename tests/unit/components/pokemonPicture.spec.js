import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture";

describe('Pruebas del componente PokemonPicture', () => {
    test('Debe de hacer prueba con el snapshot ', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 3,
                showPokemon: false
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    });

    test('Debe de mostrar la imagen oculta y el pokemon 100', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })
        const [img1, img2] = wrapper.findAll('img')

        expect(img1.exists()).toBeTruthy() 
        expect(img1.classes()).toContain('hidden-pokemon')
        expect(img2).toBe(undefined)

        expect(wrapper.props().pokemonId).toBe(100) 
        expect(wrapper.html()).toMatchSnapshot()
    });

    test('Debe de mostrar si el showPokemon es true', () => {
                const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })
        const [img1] = wrapper.findAll('img')

        expect(img1.exists()).toBeTruthy()
        expect(img1.classes()).toContain('fade-in')
        expect(img1.attributes().src).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')

    });
});