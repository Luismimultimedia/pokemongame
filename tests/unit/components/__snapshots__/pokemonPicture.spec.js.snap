// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente PokemonPicture Debe de hacer prueba con el snapshot  1`] = `<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/3.svg" alt="pokemon"></div>`;

exports[`Pruebas del componente PokemonPicture Debe de mostrar la imagen oculta y el pokemon 100 1`] = `<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg" alt="pokemon"></div>`;
